lib/portage/dbapi/porttree.py

```diff
@@ -741,7 +748,11 @@ def async_aux_get(self, mycpv, mylist, mytree=None, myrepo=None, loop=None):
             future.set_exception(PortageKeyError(mycpv))
             return future
 
-        mydata, ebuild_hash = self._pull_valid_cache(mycpv, myebuild, mylocation)
+        # If PORTDIR_OVERLAY is set, we allow checking for modifications because
+        # this happens when ebuild(1) is pointed at another location for a repository.
+        mydata, ebuild_hash = self._pull_valid_cache(
+            mycpv, myebuild, mylocation, force=lambda: "PORTDIR_OVERLAY" in os.environ
```

Comment:
The force parameter should be a plain bool rather than a lambda. Maybe we don't need the force parameter here though, since RepoConfigLoader already sets volatile to True here:
```python
                            # If PORTDIR_OVERLAY is set, we have to require volatile,
                            # because it'll break changes e.g. with ebuild(1) or
                            # development in a local repository with the same repo_name.
                            if k == "volatile" and portdir_overlay:
                                v = True
```